frontend/src/metabase/visualizations/components/settings/ChartSettingFieldPicker.jsx

```diff
@@ -69,52 +73,109 @@ const ChartSettingFieldPicker = ({
-            onChangeSeriesColor(seriesKey, value);
-          }}
-        />
-      )}
       <ChartSettingSelect
-        value={value}
+        data-testid="chartsettings-field-picker-select"
+        pl="xs"
+        pr="xs"
+        w="100%"
+        isInitiallyOpen={autoOpenWhenUnset && value === undefined}
         options={options}
+        value={value}
         onChange={onChange}
-        placeholder={t`Select a field`}
+        icon={
+          showDragHandle || (showColorPicker && seriesKey) ? (
+            <Group noWrap spacing="xs" pl="sm" pr="xs">
+              {showDragHandle && (
+                <GrabberHandle
+                  name="grabber"
+                  noMargin
+                  onClick={e => e.stopPropagation()}
+                  c="text-medium"
+                  className={CS.pointerEventsAll}
+                />
+              )}
+              {showColorPicker && seriesKey && (
+                <ChartSettingColorPicker
+                  pillSize="small"
+                  value={colors[seriesKey]}
+                  onChange={value => {
+                    onChangeSeriesColor(seriesKey, value);
+                  }}
+                  className={CS.pointerEventsAll}
+                />
+              )}
+            </Group>
+          ) : null
+        }
         placeholderNoOptions={t`No valid fields`}
-        isInitiallyOpen={autoOpenWhenUnset && value === undefined}
-        hiddenIcons
+        placeholder={t`Select a field`}
+        iconWidth="auto"
+        rightSectionWidth="auto"
+        rightSection={
+          <Group noWrap spacing="sm" p="xs" mr="sm">
+            {!disabled && (
+              <ActionIcon c="text-medium" size="sm" radius="xl" p={0}>
+                <Icon name="chevrondown" />
+              </ActionIcon>
+            )}
+            {menuWidgetInfo && (
+              <ChartSettingActionIcon
+                icon="ellipsis"
+                data-testid={`settings-${value}`}
+                onClick={e => {
+                  onShowWidget(menuWidgetInfo, e);
+                }}
+              />
+            )}
+            {onRemove && (
+              <ChartSettingActionIcon
+                icon="close"
+                data-testid={`remove-${value}`}
+                onClick={onRemove}
+              />
+            )}
+          </Group>
+        }
+        styles={{
+          wrapper: {
+            display: "flex",
+          },
+          icon: {
+            position: "static",
+            width: "auto",
+          },
+          input: {
+            "&[data-with-icon]": {
+              paddingLeft: 0,
+            },
+            marginLeft: theme.spacing.xs,
+            textOverflow: "ellipsis",
+            fontWeight: "bold",
+            "&[data-disabled]": {
+              backgroundColor: "var(--mb-color-bg-white) !important",
+            },
+            border: "none",
+            width: "100%",
+            lineHeight: theme.lineHeight,
```

Comment:
could we use `"inherit"` here, or does the parent has a different line height from the theme?

Reply:
I think I can actually remove this one - I'll see if I can remove anything else